src/refract/Element.h

```diff
@@ -0,0 +1,444 @@
+//
+//  refract/Element.h
+//  librefract
+//
+//  Created by Jiri Kratochvil on 18/05/15.
+//  Copyright (c) 2015 Apiary Inc. All rights reserved.
+//
+#if !defined _REFRACT_ELEMENT_H_
+#define _REFRACT_ELEMENT_H_
+
+#include <stdio.h>
+
+#include <string>
+#include <map>
+#include <vector>
+
+#include <sstream>
+#include <iostream>
+
+#include <stdexcept>
+
+#include "Typelist.h"
+#include "VisitableBy.h"
+
+#include "sos.h"
+
+namespace refract
+{
+
+    // Forward declarations of Elements
+    struct StringElement;
+    struct NullElement;
+    struct NumberElement;
+    struct BooleanElement;
+    struct ArrayElement;
+    struct ObjectElement;
+    struct MemberElement;
+
+
+    // Forward declarations of Visitors
+    // if you want add new one, do not forget add IElement::Visitors
+    struct ComparableVisitor;
+    struct SerializeVisitor;
+    struct SerializeCompactVisitor;
+
+    struct LogicError : std::logic_error {
+        explicit LogicError(const std::string& msg) : std::logic_error(msg) {}
+    };
+    // will be removed in future
+    struct NotImplemented : std::runtime_error {
```

Comment:
Please add empty lines before new structs. Improves readability and maintainability of the code.